src/os/bluestore/BlueStore.cc

```diff
@@ -663,12 +665,12 @@ void BlueStore::BufferSpace::read(
   res.clear();
   uint64_t end = offset + length;
   for (auto i = _data_lower_bound(offset);
-       i != buffer_map.end() && offset < end && i->first < end;
+       i != buffer_map.end() && offset < end && i->first < end && length;
```

Comment:
No need for that addition - the length == 0 case is properly handled by offset < end